clans/src/main/java/me/mykindos/betterpvp/clans/weapons/impl/cannon/listener/CannonListener.java

```diff
@@ -145,7 +153,16 @@ public void onCustomDamage(final PreCustomDamageEvent pre) {
             final Player player = Bukkit.getPlayer(originalOwner);
             if (player != null) {
                 event.setDamager(player);
-                event.setDamage(cannonballDamage);
+                double distance = tnt.getLocation().distance(event.getDamagee().getLocation());
+                double damage;
+                if (distance < cannonballDamageMinRadius) {
+                    damage = cannonballDamage;
+                } else if (distance > cannonballDamageMaxRadius) {
+                    damage = 0;
+                } else {
+                    damage = cannonballDamage * (distance - cannonballDamageMinRadius) / cannonballDamageMaxRadius - cannonballDamageMinRadius;
+                }
+                event.setDamage(damage);
```

Comment:
<= and >=, what if they are exactly 4 blocks away

If cannon damage is 15, and a player is 3.99 blocks away
15 * (3.99 - 1) / 4 - 1 = 10.2 damage, i feel like this could be simplified a bit and maybe reduce damage by more depending on distance

Reply:
Math was not mathing, math was fixed. <= and >= is not necessary, as the scaling function works on those numbers.

Reply:
![image](https://github.com/Mykindos/BetterPvP/assets/5389944/8c286d1b-6890-4679-a5d1-854d3ceb2af6)

I would be ok with this, but we probably want to deal at least some damage if they were caught in the radius

Reply:
What if it was damage = Math.max(cannonballDamage * ((deltaRadius - (distance - cannonballDamageMinRadius)) / deltaRadius), 1);

Reply:
at max range, you'd at least deal 1 damage

Reply:
lets make the minimum 4 (2 hearts to a naked)

Reply:
implemented